Parent/Parent/View/CustomNavbarProtocol.swift

```diff
@@ -206,7 +207,18 @@ extension CustomNavbarProtocol {
         navigationController?.navigationBar.barTintColor = customNavBarColor
         navbarBottomViewContainer.backgroundColor = customNavBarColor
     }
+
+    var isCustomNavMenuEnabled: Bool {
+        get {
+            let enabled = objc_getAssociatedObject(self, &customNavBarMenuEnabledKey) as? Bool
+            return enabled ?? true
+        }
+        set {
+            objc_setAssociatedObject(self, &customNavBarMenuEnabledKey, newValue, .OBJC_ASSOCIATION_ASSIGN)
```

Comment:
I'm not a huge fan of these associated objects in a Swift world. What would you think about making this check a method on the protocol with a default implementation?

```swift
protocol CustomNavbarProtocol {
  func shouldShowCustomNavbarMenu() -> Bool
}

extension CustomNavbarProtocol {
  func shouldShowCustomNavbarMenu() -> Bool {
    return true
  }

  func showCustomNavbarMenu(_ show: Bool = true, completion: (() -> Void)? = nil) {
    guard show && shouldShowCustomNavbarMenu() else { return }
    ...
  }
}

extension DashboardPresenter {
  func shouldShowCustomNavbarMenu() -> Bool {
    return students.count > 1
  }
}
```
Your thoughts?

Reply:
I was looking at the next ticket to work on https://instructure.atlassian.net/browse/MBL-13895 and i don't think we'll ever actually need this b/c if there is only 1 student, we would still show `add student` button so we would never need to disable the menu from showing.  How do you feel we leave this as is, and I quickly follow up with https://instructure.atlassian.net/browse/MBL-13895 which will remove this?

Reply:
Sure.

Reply:
...leave this as is, and merge.  Quickly follow up in a new pr with https://instructure.atlassian.net/browse/MBL-13895